src/main/scala/com/yotpo/metorikku/metric/Metric.scala

```diff
@@ -74,6 +74,36 @@ case class Metric(configuration: Configuration, metricDir: File, metricName: Str
     }
   }
 
+  private def writeToHive(job: Job, writer: Writer, outputConfig: Output) = {
```

Comment:
why its not a part of the writer factory? why we have write and write to hive methods?

Reply:
It's not part of a single writer. any writer can support this feature. so it should be outside of the writer.
It falls under the metric concern in my opinion.